test/e2e/vmagent_test.go

```diff
@@ -0,0 +1,327 @@
+		t.Errorf("Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifIndex, stderr)
+	}
+
+	vm.ifIndex = strings.TrimSpace(ifIndex)
+	cmd = fmt.Sprintf("powershell 'Get-NetAdapter -IfIndex %s | select name | ft -HideTableHeaders'", vm.ifIndex)
+	rc, ifName, stderr, err := data.RunCommandOnNode(nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, nodeName, err)
+	if rc != 0 {
+		t.Errorf("Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifName, stderr)
+	}
+
+	vm.ifName = strings.TrimSpace(ifName)
+	cmd = fmt.Sprintf("powershell 'Get-NetIPAddress -AddressFamily IPv4 -ifIndex %s| select IPAddress| ft -HideTableHeaders'", vm.ifIndex)
+	rc, ifIP, stderr, err := data.RunCommandOnNode(nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, nodeName, err)
+	if rc != 0 {
+		t.Errorf("Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, ifIP, stderr)
+	}
+	vm.ip = strings.TrimSpace(ifIP)
+	return vm
+
+}
+
+func startAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Starting antrea-agent on VM: %s", vm.nodeName)
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = "nssm start antrea-agent"
+	} else {
+		cmd = "sudo systemctl start antrea-agent"
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	if rc != 0 {
+		t.Errorf("Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, stdout, stderr)
+	}
+}
+
+func stopAntreaAgent(t *testing.T, data *TestData, vm vmInfo) {
+	t.Logf("Stopping antrea-agent on VM: %s", vm.nodeName)
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = "nssm stop antrea-agent"
+	} else {
+		cmd = "sudo systemctl stop antrea-agent"
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	require.NoError(t, err, "Failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	if rc != 0 {
+		t.Errorf("Failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, stdout, stderr)
+	}
+}
+
+func verifyInterfaceIsInOVS(t *testing.T, data *TestData, vm vmInfo) (found bool, err error) {
+	var cmd string
+	if vm.osType == "Windows" {
+		cmd = fmt.Sprintf("ovs-vsctl --column=name list port '%s'", vm.ifName)
+	} else {
+		cmd = fmt.Sprintf("sudo ovs-vsctl --column=name list port %s", vm.ifName)
+	}
+	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
+	if err != nil {
+		return false, fmt.Errorf("failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
+	}
+	if rc != 0 {
+		return false, fmt.Errorf("failed to run command: <%s>, stdout: <%v>, stderr: <%v>", cmd, stdout, stderr)
+	}
+
+	if strings.Contains(stdout, "no row") {
+		return false, fmt.Errorf("failed to find OVS port %s on VM %s, err %v", vm.ifName, vm.nodeName, err)
+	}
+
+	if strings.Contains(stdout, vm.ifName) && strings.Contains(stdout, "name") {
+		return true, nil
+	}
```

Comment:
when `ovs-vsctl --column=name list port` specifies port name, I think `rc` already represent whether the port exists? The following two checks seems useless

Reply:
We are interested only in the below outputs and it should retry for any other  output (irrespective of status code).

``` 
ovs-vsctl: no row "ens192" in table Port --> failure case
name                : eth0 --> success case
```


So I'm changing this into.
```
	rc, stdout, stderr, err := data.RunCommandOnNode(vm.nodeName, cmd)
	if err != nil {
		return false, fmt.Errorf("failed to run command <%s> on VM %s, err %v", cmd, vm.nodeName, err)
	}

	if strings.Contains(stdout, "no row") {
		return false, fmt.Errorf("failed to find OVS port %s on VM %s, err %v, rc %d, stdout %v, stderr %v", vm.ifName, vm.nodeName, err, rc, stdout, stderr)
	}

	if strings.Contains(stdout, vm.ifName) && strings.Contains(stdout, "name") {
		return true, nil
	}
```